src/parser/creative_linear_parser.js

```diff
@@ -158,6 +159,30 @@ export function parseCreativeLinear(creativeElement, creativeAttributes) {
 
           creative.mediaFiles.push(mediaFile);
         });
+
+      const mezzanineElement = parserUtils.childByName(mediaFilesElement, 'Mezzanine');
+      if (mezzanineElement) {
+        const mezzanine = new Mezzanine();
+        mezzanine.id = mezzanineElement.getAttribute('id');
+        mezzanine.fileURL = parserUtils.parseNodeText(mezzanineElement);
+        mezzanine.delivery = mezzanineElement.getAttribute('delivery');
+        mezzanine.codec = mezzanineElement.getAttribute('codec');
+        mezzanine.type = mezzanineElement.getAttribute('type');
+        mezzanine.width = parseInt(
+            mezzanineElement.getAttribute('width') || 0, 10
```

Comment:
I'm not a big fan of having default value at 2 places, here you still put 0 as default value but inside the constructor of `class Mezzanine` you also put 0 as default value. I'm asking do we really need to do the `validate` on the class level ? shouldn't we do an early return as soon as we parsed the Mezzanine Element ?
```suggestion
            if (mezzanineElement && mandatoryFields1 !== null etc ...) {
```
Instead of instantiating `new Mezzanine()`to discard the Mezzanin ?
WDYT ?

Reply:
we have 4 mandatory fields,
And
if(!field1 || !field2 || !field3 || !field4) return
is ugly
parseInt(null, 10) would give me NaN and i dont want to check 3 different values (null, 0, NaN) so I stick with 0

Logically everything is fine I hope, and here we discuss just style preferences, I would like to avoid it. Any suggestions?

Reply:
I will be taking over Rumesh as he is off. What he meant is that you should first parse the required fields (without instantiating the class), and then call a function to check if those required fields are valid. If they are valid, you can instantiate the class. If they aren't, we should discard the Mezzanine Element

Reply:
The reason why we want this is because we are moving away from classes. Initially they were only used as a way of documentation, they aren't necessary anymore.